jme3-core/src/main/java/com/jme3/anim/tween/action/BlendAction.java

```diff
@@ -1,129 +1,265 @@
+ * using the blendWeight as a delta value, then interpolating the resulted value with the current target local transforms,
+ * and finalizing by applying the final collected result of interpolation to the target as a local transform matrix.
+ *
+ * If the blendWeight is :
+ * - less than 1, then interpolate the 1st action & the second one & apply the results.
+ * - equal to zero, then interpolate the 1st action only & apply the results.
+ * - equal to one, then interpolate only the second action with blendWeight = 1.
+ *
+ * <b>Created by Nehon.</b>
+ */
 public class BlendAction extends BlendableAction {
 
     private int firstActiveIndex;
     private int secondActiveIndex;
     final private BlendSpace blendSpace;
-    private float blendWeight;
-    final private double[] timeFactor;
+    private double[] timeFactor;
     final private Map<HasLocalTransform, Transform> targetMap = new HashMap<>();
 
+    /**
+     * Creates an action that utilizes a blend space to blend between 2 selected actions from the collection of actions provided.
+     * @param blendSpace the blendSpace, used for calculating the blendWeight
+     * @param actions the collection of actions, used for selecting 2 actions to blend between them.
+     */
     public BlendAction(BlendSpace blendSpace, BlendableAction... actions) {
         super(actions);
-        timeFactor = new double[actions.length];
         this.blendSpace = blendSpace;
         blendSpace.setBlendAction(this);
 
-        for (BlendableAction action : actions) {
+        //prepare for interpolating actions by gathering targets & timeFactors based on the actionLength & the global length.
+        gatherTargetsFromActions(actions);
+        gatherTimeFactorPerAction(actions);
+    }
+
+    /**
+     * Used for gathering the timeFactor or the timeDelta value for the inscribed actions.
+     * The timeFactor : is used to scale the tpf for the current running inscribed action.
+     * Each action should have its own timeFactor used for interpolation.
+     * @param actions the actions, used for fetching the timeFactor values.
+     */
+    private void gatherTimeFactorPerAction(final BlendableAction... actions){
+        timeFactor = new double[actions.length];
+        for (int i = 0; i < actions.length; i++) {
+            //get the action duration.
+            final double actionLength = actions[i].getLength();
+            if (actionLength > 0 && getLength() > 0) {
+                //get the timeFactor(timeScale) of the inscribed action with respect to its parent Action.
+                this.timeFactor[i] = actionLength / getLength();
+            }else{
+                //assign a default value if durations are negative
+                this.timeFactor[i] = 1;
+            }
+        }
+    }
+
+    /**
+     * Used for gathering the targets & prepare them from the parsed anim actions args.
```

Comment:
Parsing refers to creating a hierarchy from a string of text or tokens. This method doesn't parse anything.

This javadoc obscures what the method actually does and doesn't say much that isn't conveyed by the menthod name. It might be better to delete the javadoc.

Reply:
I think keeping javadoc for a private method won't hurt after all javadoc is for us is for reading only, can i just rephrase it and keep it ?

Reply:
Yes, if you can rephrase it to be accurate and not obscure.